service/view_building_coordinator.cc

```diff
@@ -139,6 +147,139 @@ future<std::optional<view_building_coordinator::vbc_state>> view_building_coordi
     co_return state;
 }
 
+static std::optional<dht::token_range> get_range_to_build(const locator::tablet_map& tablet_map, const dht::token_range_vector ranges) {
+    for (auto& range: ranges) {
+        auto tid = tablet_map.get_tablet_id(range.end().value().value());
+        auto trinfo = tablet_map.get_tablet_transition_info(tid);
+
+        if (!trinfo) {
+            return range;
+        }
+    }
```

Comment:
We need to mutually exclude migration and view building. If a range is being built, it can't be migrated.

Reply:
We decided that migration is more important. If a range is being built and migration starts, the building work is aborted and started again once the migration is finished.